### PR TITLE
feat: object-based edits

### PR DESCRIPTION
**Pull request by aidenybai**
Here is a somewhat different approach that uses iteration but tries to monomorphically use objects instead of arrays

### CODE REVIEW

File: packages/million/dom.ts [hunk 1]
```diff
@@ -158,7 +147,7 @@ export const setAttribute = (
     el[name] !== undefined &&
     el[name] !== null &&
     !(el instanceof SVGElement) &&
-    setHas$.call(NON_PROPS, name)
+    NON_PROPS.has(name)
```
Comment (SukkaW) [inline packages/million/dom.ts:150]: We should keep the prototype cache here. Prototype cache can skip the prototype chain lookup.

Comment (SukkaW) [reply packages/million/dom.ts:150]: **Chrome**

image

**Firefox**

image

Comment (aidenybai) [reply packages/million/dom.ts:150]: good catch!

File: packages/compiler/visitor.ts [hunk 2]
```diff
@@ -209,3 +237,137 @@ export const visitCallExpression = (path: NodePath<t.CallExpression>) => {
     }
   }
 };
+
```
Comment (aidenybai) [inline packages/compiler/visitor.ts:240]: marker

**Pull request merged**